tests/e2e/tests/e2e_util/util.py

```diff
@@ -57,11 +56,120 @@ def run_with_retry(
                     raise e
 
 
+def retry_until_condition(
```

Comment:
seems like we start to need more advanced retry logic. 
It's probably time to move to something like https://github.com/jd/tenacity

Reply:
no need to do this in this PR - just add a TODO so that we can create follow-up cards